### DISCUSSION TITLE
Option to Suppress existing logs

### DISCUSSION THREAD

**Discussion by shreyaUp** [apache/airflow#21270] [Q&A]
### Description

Hi,

Would it be possible to suppress airflow existing logs for task instance and use custom logging? 

### Use case/motivation

The main reason behind this is to suppress the logging of connection details that is printed without masking the Host and Port details. (XX are added by me below)

`[2022-01-24, 19:10:48 UTC] {base.py:70} INFO - Using connection to: id: mysql_default. Host: 10.22.X.X, Port: 3306, Schema: XX, Login: XX, Password: ***, extra: {}`

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
See https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/logging-tasks.html#advanced-configuration

PLease do not open "issues" for questions like that. If you are unsure and ask question you can open discussion instead 
 https://github.com/apache/airflow/discussions

**Reply by shreyaUp** [reply]
Thanks @potiuk. Is it possible to use the log_config file, with additional filter added, if we do not wish to perform remote logging but instead wish to perform log filtering to mask certain values in the DAG logs?

**Reply by potiuk** [reply]
Yes. But for masking, you probably want to use the built-in masking https://airflow.apache.org/docs/apache-airflow/stable/security/secrets/mask-sensitive-values.html

**Reply by shreyaUp** [reply]
I have tried adding values to sensitive_var_conn_names for masking but for some reason, it doesn't work. In the above link, it says 
_"If you want to mask an additional secret that is already masked by one of the above methods.. "_  
Should it be that is not already masked or do we need to specify it twice, once in the ENV var and then again in the code?

**Reply by potiuk** [reply]
Maybe the reason is that masking only happens in tasks. if you want to mask it elsewhere, Airflow is not doing that (as in the standard setup it is not needed to mask anything in scheduler or webserver or worker low-level logs - only task logs are visible in the UI.

You can look for this in Airflow Code: 

```
    settings.MASK_SECRETS_IN_LOGS = True
```

Assuming that you want to mask things elsewhere, you need to set those settings to True (either globally or specifically in the python code you want to run). But this is not a standard feature of Airflow and it might or might not work and you might or might not modify your logging configuration in the way described in https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/logging-tasks.html#advanced-configuration

Airflow uses standard Python logging and default configuration provided handles the case above - but there is nothing preventing you to modify the logging as you want. Airlfow uses standard Python Logging facilities that allow for infinite configurability https://docs.python.org/3/library/logging.html

**Reply by josh-fell** [reply]
@shreyaUp FWIW that particular logging line will be pushed to the DEBUG level as part of #21162 so hopefully any headaches you have about log masking of connection details dissipate in a future release.

**Reply by shreyaUp** [reply]
@potiuk My concern was exactly that @josh-fell has taken care of in https://github.com/apache/airflow/pull/21162. However, since I am using a managed Airflow environment, don't really know when they will upgrade Airflow version.

For now, I am trying to achieve the same by changing logging config by the method suggested in https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/logging-tasks.html#advanced-configuration. I guess I am missing something since it's not getting reflected.

Changes I am making:
1. Changing logging_config_class = log_config.LOGGING_CONFIG but not remote logging since I won't be using that.
2. Contents of log_config.py
```
from copy import deepcopy
import logging
from logging.config import dictConfig
from airflow.config_templates.airflow_local_settings import DEFAULT_LOGGING_CONFIG
import airflow.logging_config
log = logging.getLogger(__name__)

class MyFilter(logging.Filter):
    def __init__(self, param=None):
        self.param = param

    def filter(self, record):
        if self.param is None:
            allow = True
        else:
            allow = self.param not in record.msg
        if allow:
            record.msg = 'changed: ' + record.msg
        return allow


LOGGING_CONFIG = deepcopy(DEFAULT_LOGGING_CONFIG)
log.info("Logger: {}",LOGGING_CONFIG)
LOGGING_CONFIG = LOGGING_CONFIG['filters']['sensitive'] = {
            '()': MyFilter,
            'param': 'custom_value',
        }
LOGGING_CONFIG['handlers']['console']['filters'].append('sensitive')
LOGGING_CONFIG['handlers']['task']['filters'].append('sensitive')
LOGGING_CONFIG['handlers']['processor']['filters'].append('sensitive')

LOGGING_CONFIG['loggers']['airflow.task']['filters'].append('sensitive')

LOGGING_CONFIG['root']['filters'].append('sensitive')
airflow.logging_config.dictConfig(LOGGING_CONFIG)
```

**Reply by potiuk** [reply]
I am not sure if that will work in managed environment. There are some limitations in a number of such managed environments, and I think it is the best to check with them.  Being able to modify logging seems like might be one of likely limitations - I imagine any of such managed environments already modifies Airflow logging to pre-integrate it with their chosen logging solutions (CloudWatch for MWAA, Stackdriver for GCP etc)., so I would not be surprised at all if this is very different than default airlfow logging configuration and would not be surprised at all if the configuration cannot be modified or has some heavy limitations.

This is part of the price you pay for managed environment that you get parts of the system "managed" for you but you loose certain "flexibilities" (precisely because it is managed for you). What I can suggest - you can try the logging configuration of yours by following one of the Airflow "quick-starts" - using "vanilla" local airflow deployment. If you will make it works there and it will not work in tha managed environment, I suggest you raise an issue with them.
